server/startup/restApiEndpoints.js

```diff
@@ -0,0 +1,183 @@
+import collections from "../../lib/collections";
+import Reaction from "../../server/api/core";
+
+/**
+ * @param {Object} user
+ * @param {String} role
+ * @returns {boolean}
+ */
+const hasPermission = (user, role) => {
+  return user.roles[Reaction.getShopId()].includes(role);
+};
+
+// eslint-disable-next-line no-undef
+const Api = new Restivus({
+  useDefaultAuth: true,
+  prettyJson: true,
+  version: "v1",
+  defaultHeaders: {
+    "Content-Type": "application/json"
+  }
+});
+
+/**
+ * @param {String} collectionName - Name of collection
+ * @param {Object} collection - collection Schema
+ * @param {Object} restApi - object
+ * @returns none
+ */
+const createRestApiFor = (collectionName, collection, restApi = Api) => {
+  restApi.addCollection(collection, {
+    path: collectionName,
+    routeOptions: { authRequired: true },
+    endpoints: {
+      getAll: {
+        authRequired: true,
+        action() {
+          if (hasPermission(this.user, "admin") || hasPermission(this.user, "guest") || hasPermission(this.user, "owner")) {
+            const all = collection.find().fetch();
+            if (!all) {
+              return {
+                status: "failed",
+                statusCode: 404,
+                message: `No ${collectionName} found!`
+              };
+            }
+            return {
+              status: "success",
+              statusCode: 200,
+              data: {
+                all
+              }
+            };
+          }
+          return {
+            statusCode: 403, status: "failed",
+            message: "You do not have permission to get a record"
+          };
+        }
+      },
+      post: {
+        authRequired: true,
+        action() {
+          if (hasPermission(this.user, "admin") || hasPermission(this.user, "owner")) {
+            let error = false;
+            let id;
+            if (this.bodyParams) {
+              collection.insert(this.bodyParams, (err, _id) => {
+                if (err) {
+                  error = true;
+                }
+                id = _id;
+              });
+              return error ? {
+                status: "Failed",
+                statusCode: 500,
+                message: `unable to add ${collectionName}`
+              } : {
+                status: "success",
+                statusCode: 201,
+                data: {
+                  id
+                }
+              };
+            }
+          }
+          return {
+            statusCode: 403, status: "failed",
+            message: "You do not have permission to add a record"
+          };
+        }
+      },
+      get: {
+        authRequired: true,
+        action() {
+          if (hasPermission(this.user, "admin") || hasPermission(this.user, "guest") || hasPermission(this.user, "owner")) {
+            const foundItems = collection.findOne(this.urlParams.id);
+            if (foundItems !== undefined) {
```

Comment:
This could just be 
```js
if (!foundItems)
```
undefined is not the only falsy value in javascript, hence this code could return success  even if nothing was found